programs/eosio-blocklog/main.cpp

```diff
@@ -169,6 +176,176 @@ void blocklog::initialize(const variables_map& options) {
+   constexpr uint32_t bufLen{1U<<24};                       //bufLen must be power of 2 >= largest possible block == one MByte
+   char* buf= new char[bufLen+8];                           //first 8 bytes of prior buf are put past end of current buf
+   constexpr uint64_t fposListLen{1U<<22};                  //length of fposList[] in bytes
+   uint64_t* fposList= new uint64_t[fposListLen>>3];
+
+   //read blocks.log to see if version 1 or 2 and get firstblocknum (implicit 1 if version 1)
+   uint32_t version=0, firstBlock;
+   read(fin,(char*)&version,sizeof(version));
+   EOS_ASSERT( version==1 || version==2, block_log_unsupported_version, "unsupported version of block log" );
+   if (version == 1)
+      firstBlock= 1;
+   else
+      read(fin,(char*)&firstBlock,sizeof(firstBlock));
+   cout << "block log version= " << version << '\n';
+   cout << "first block= " << firstBlock << '\n';
+
+   uint64_t pos= lseek(fin,0,ios::end);                     //get blocks.log file length
+   uint64_t lastBufLen= pos & ((uint64_t)bufLen-1);         //bufLen is a power of 2 so -1 creates low bits all 1
+   if (!lastBufLen)                                         //will read integral number of bufLen and one time read lastBufLen
+      lastBufLen= bufLen;
+   pos= lseek(fin,-(uint64_t)lastBufLen,ios::end);
+   uint64_t didread= read(fin,buf,lastBufLen);              //read tail of file into buf
+   EOS_ASSERT( didread==lastBufLen, block_log_exception, "blocks.log read fails" );
+
+   uint32_t indexStart;                                     //buf index for block start
+   uint32_t indexEnd;                                       //buf index for block end == buf index for block start file position
+   uint64_t filePos;                                        //file pos of block start
+   BlockStart* bst;                                         //pointer to BlockStart
+
+   indexStart= lastBufLen;                                  //pretend a block starts just past end of buf then read prior block
+   indexEnd= indexStart-8;                                  //index in buf where block ends and block file position starts
+   filePos= *(uint64_t*)(buf+indexEnd);                     //file pos of block start
+   indexStart= filePos - pos;                               //buf index for block start
+   bst= (BlockStart*)(buf+indexStart);                      //pointer to BlockStart
+   uint32_t lastBlock= endian_reverse_u32(bst->blknum);     //convert from big endian to little endian
+   uint32_t bnum= ++lastBlock;                              //add 1 since took block number from prior block id
+   cout << "last block=  " << lastBlock << '\n';
+   cout << '\n';
+   cout << "block " << setw(10) << bnum << "    filePos " << setw(14) << filePos << '\n';  //first progress indicator
+
+   //we use low level file IO because it is distinctly faster than C++ filebuf or iostream
```

Comment:
If we really have to POSIXify, we probably should `#ifndef _WIN32` the pieces that won't compile/work on win32